azurerm/data_source_api_management.go

```diff
@@ -0,0 +1,336 @@
+		d.Set("gateway_regional_url", props.GatewayRegionalURL)
+		d.Set("portal_url", props.PortalURL)
+		d.Set("management_api_url", props.ManagementAPIURL)
+		d.Set("scm_url", props.ScmURL)
+		d.Set("static_ips", props.PublicIPAddresses)
+
+		if err := d.Set("hostname_configuration", flattenDataSourceApiManagementHostnameConfigurations(props.HostnameConfigurations)); err != nil {
+			return fmt.Errorf("Error setting `hostname_configuration`: %+v", err)
+		}
+
+		if err := d.Set("additional_location", flattenDataSourceApiManagementAdditionalLocations(props.AdditionalLocations)); err != nil {
+			return fmt.Errorf("Error setting `additional_location`: %+v", err)
+		}
+	}
+
+	if err := d.Set("sku", flattenDataSourceApiManagementServiceSku(resp.Sku)); err != nil {
+		return fmt.Errorf("Error flattening `sku`: %+v", err)
+	}
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func flattenDataSourceApiManagementHostnameConfigurations(input *[]apimanagement.HostnameConfiguration) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	// management, portal, proxy, scm
+	managementResults := make([]interface{}, 0)
+	proxyResults := make([]interface{}, 0)
+	portalResults := make([]interface{}, 0)
+	scmResults := make([]interface{}, 0)
+
+	for _, config := range *input {
+		output := make(map[string]interface{}, 0)
+
+		if config.HostName != nil {
+			output["host_name"] = *config.HostName
+		}
+
+		if config.NegotiateClientCertificate != nil {
+			output["negotiate_client_certificate"] = *config.NegotiateClientCertificate
+		}
+
+		if config.KeyVaultID != nil {
+			output["key_vault_id"] = *config.KeyVaultID
+		}
+
+		switch strings.ToLower(string(config.Type)) {
+		case strings.ToLower(string(apimanagement.Proxy)):
+			// only set SSL binding for proxy types
+			if config.DefaultSslBinding != nil {
+				output["default_ssl_binding"] = *config.DefaultSslBinding
+			}
+			proxyResults = append(proxyResults, output)
+			break
+
+		case strings.ToLower(string(apimanagement.Management)):
+			managementResults = append(managementResults, output)
+			break
+
+		case strings.ToLower(string(apimanagement.Portal)):
+			portalResults = append(portalResults, output)
+			break
+
+		case strings.ToLower(string(apimanagement.Scm)):
+			scmResults = append(scmResults, output)
+			break
+		}
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"management": managementResults,
+			"portal":     proxyResults,
+			"proxy":      portalResults,
+			"scm":        scmResults,
+		},
+	}
+}
+
+func flattenDataSourceApiManagementAdditionalLocations(input *[]apimanagement.AdditionalLocation) []interface{} {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results
+	}
+
+	for _, prop := range *input {
+		output := make(map[string]interface{}, 0)
+
+		if prop.Location != nil {
+			output["location"] = azureRMNormalizeLocation(*prop.Location)
+		}
+
+		if prop.PublicIPAddresses != nil {
+			output["static_ips"] = *prop.PublicIPAddresses
```

Comment:
on reflection, I guess this should probably be `public_ip_addresses` (since `private_ip_addresses` can also be returned for API Management Services on a network)?

Reply:
Renamed to `public_ip_addresses`. Just need to remember to add `private_ip_addresses` when/if we add support for internal vnets. I will retest this (internal vnets) now that we use the preview API, but I expect the same problem to be present.